### ISSUE TITLE
cmake compile error with GNURadio master

### ISSUE THREAD

**Issue by xmpuri** [bastibl/gr-foo#24]
Hi,
I am getting the following compile errors
```
cmake ..
-- Build type not specified: defaulting to release.
CMake Warning at CMakeLists.txt:84 (find_package):
  By not providing "FindUHD.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "UHD", but
  CMake did not find one.

  Could not find a package configuration file provided by "UHD" with any of
  the following names:

    UHDConfig.cmake
    uhd-config.cmake

  Add the installation prefix of "UHD" to CMAKE_PREFIX_PATH or set "UHD_DIR"
  to a directory containing one of the above files.  If "UHD" provides a
  separate development package or SDK, be sure it has been installed.


-- UHD library not found, no support for timed TX.
CMake Error at CMakeLists.txt:98 (find_package):
  Could not find a configuration file for package "Gnuradio" that is
  compatible with requested version "3.8".

  The following configuration files were considered but not accepted:

    /usr/lib/x86_64-linux-gnu/cmake/gnuradio/GnuradioConfig.cmake, version: 3.7.9.1



-- Configuring incomplete, errors occurred!
See also "/home/rp/gr-foo/build/CMakeFiles/CMakeOutput.log".

```
I installed GNU Radio from prebuilt ppa: https://wiki.gnuradio.org/index.php/InstallingGR#Linux
and I followed the commands in your readme https://github.com/bastibl/gr-foo#installation

Please advise on what I should do to fix it.
I'm trying to run your code from your other projects: https://github.com/bastibl/gr-ieee802-11

**Comment by bastibl** [comment]
You use incompatible branches. It's the second sentence of the readme and was already addressed in several issues.

https://github.com/bastibl/gr-foo/issues/13
https://github.com/bastibl/gr-foo/issues/14
https://github.com/bastibl/gr-foo/issues/8
https://github.com/bastibl/gr-foo/issues/6

**Comment by xmpuri** [comment]
Sorry I realize now that the default branch of this repo was not master.

I've now reached a new issue in the make process:
```Scanning dependencies of target gnuradio-foo
[  7%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/burst_tagger_impl.cc.o
[ 15%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/channel_model_impl.cc.o
[ 23%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_dropper_impl.cc.o
[ 30%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_pad2.cc.o
[ 38%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_pad_impl.cc.o
[ 46%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/periodic_msg_source_impl.cc.o
[ 53%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/random_periodic_msg_source_impl.cc.o
[ 61%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/rtt_measure_impl.cc.o
[ 69%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/wireshark_connector_impl.cc.o
[ 76%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/pad_tagged_stream_impl.cc.o
make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/liborc-0.4.so', needed by 'lib/libgnuradio-foo.so.69755e4f'.  Stop.
CMakeFiles/Makefile2:135: recipe for target 'lib/CMakeFiles/gnuradio-foo.dir/all' failed
make[1]: *** [lib/CMakeFiles/gnuradio-foo.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2```
Do you have any advice how to address this?

**Comment by bastibl** [comment]
So what did you install, i.e., what version of GR and this module? Make sure that you have only one installation of GR on your system. You could also make sure that `liborc` is installed, but since you seem to have a GR/Vol installation it should be there.

**Comment by redfoxbluefox** [comment]
I have this same error. I have installed both UHD and GNU Radio via apt after updating the ppa. My gnuradio (companion) version is 3.8.1.0. My system info:
```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
```

This is a very recent OS installation, so I'm certain there is only one version of GNU Radio installed.

My gr-foo is on branch maint-3.8. The 'cmake ..' command works, but some of the things it's looking for are not found: 
```
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Build type not specified: defaulting to release.
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found UHD: /usr/lib/x86_64-linux-gnu/libuhd.so  
-- UHD library found.
-- Found LOG4CPP: /usr/lib/x86_64-linux-gnu/liblog4cpp.so
-- Checking for module 'gmp'
--   No package 'gmp' found
-- Found GMP: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Checking for module 'mpir >= 3.0'
--   No package 'mpir' found
-- Could NOT find MPIR (missing: MPIRXX_LIBRARY MPIR_LIBRARY MPIR_INCLUDE_DIR) 
-- Found MPLIB: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   date_time
--   program_options
--   filesystem
--   system
--   regex
--   thread
--   unit_test_framework
--   chrono
--   atomic
-- Found VOLK: Volk::volk  
-- User set python executable /usr/bin/python3
-- Found PythonInterp: /usr/bin/python3 (found version "3.6.9") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.6m.so (found suitable exact version "3.6.9") 
-- Checking for module 'fftw3f >= 3.0'
--   Found fftw3f , version 3.3.7
-- Found FFTW3f: /usr/lib/x86_64-linux-gnu/libfftw3f.so  
-- Found Git: /usr/bin/git  
-- Extracting version information from git describe...
-- Using install prefix: /usr/local
-- Building for version: 13fac53d / 1.1.0git
-- 
-- Checking for module SWIG
-- Disabling SWIG because version check failed.
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.6m.so (found version "3.6.9") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/harper/Repos/gr-foo/build
```

Then the make step fails:
```
Scanning dependencies of target pygen_python_c5615
Scanning dependencies of target gnuradio-foo
[  7%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_pad2.cc.o
[ 15%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/periodic_msg_source_impl.cc.o
[ 23%] Generating __init__.pyc, selector.pyc
[ 30%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_dropper_impl.cc.o
[ 38%] Generating __init__.pyo, selector.pyo
[ 46%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/burst_tagger_impl.cc.o
[ 53%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/channel_model_impl.cc.o
[ 61%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/packet_pad_impl.cc.o
[ 69%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/random_periodic_msg_source_impl.cc.o
[ 69%] Built target pygen_python_c5615
[ 76%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/rtt_measure_impl.cc.o
[ 84%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/wireshark_connector_impl.cc.o
make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/liborc-0.4.so', needed by 'lib/libgnuradio-foo.so.13fac53d'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 92%] Building CXX object lib/CMakeFiles/gnuradio-foo.dir/pad_tagged_stream_impl.cc.o
CMakeFiles/Makefile2:135: recipe for target 'lib/CMakeFiles/gnuradio-foo.dir/all' failed
make[1]: *** [lib/CMakeFiles/gnuradio-foo.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

I've also checked that liborc is installed:
```
Reading package lists... Done
Building dependency tree       
Reading state information... Done
liborc-0.4-0 is already the newest version (1:0.4.28-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

Any other ideas what can be going on? 
Thanks for your help, 
Andrew

**Comment by xmpuri** [comment]
Perfect. I had to both run `sudo apt-get install liborc-0.4.0` as well as create the softlink like @rajb245 suggested.

**Comment by bastibl** [comment]
Happy it works. I just checked again and this is AFAIK not an issue of this module. It does not use ORC but uses GNU Radio, which uses VOLK, which uses ORC. I'd assume that other GNU Radio OOTs would show similar errors. It also seems really strange that you have VOLK installed but not `orc`.  It either means that the VOLK package is totally broken (both its dependencies and the path in `VolkTargets.cmake`) or that you have (leftovers from) multiple versions on your system.

**Comment by rajb245** [comment]
Perhaps the GNURadio package in the PPA is broken. Another option is that liborc is broken in the Ubuntu repos. The third option is that cmake should pick up the alternate shared object names, like "liborc-0.4.so.0", but is failing to do so due to the configuration or a bug. That there is no file named "liborc-0.4.so" is weird to me, and I reported that as a bug to Ubuntu: https://bugs.launchpad.net/ubuntu/+source/orc/+bug/1869979

I can confirm that I recreated the issue with only a single install of GNURadio: I made a new VirtualBox VM with Xubuntu 18.04 and went through the steps @redfoxbluefox outlined ONLY.

If its not a liborc issue, it needs to be investigated a bit and reported upstream to GNURadio. I guess I'll bite.
